### PR TITLE
Fix: Fixed "Size All Columns to Fit" not working properly

### LINKED ISSUE #11295: Bug: Size to fit for the tag column is broken
### Description

Trying to size the tag column to fit all the tags in the details layout is broken.

### Steps To Reproduce

1. Open the details layout
2. Add some tags to files and folders
3. Double click the grid splitter for the tag column
4. See how the column isn't resized to fit all the tags

Also
1. Right click the header and click on "size columns to fit"
2. See how the column isn't resized to fit all the tags

### Steps to fix

- Properly resize the tag column to fit the tags

### Files Version

v2.4.34

### Windows Version

Windows 11

### Log file

NA

### PR DESCRIPTION
**Pull request by mafra99**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #11295

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Note**
Probably caused by #11216
Before that PR only one tag was displayed and there was only one TextBlock

Now for each Tag a separate TextBlock is displayed. The changed in this PR should calculate the size of the complete StackPanel (FontIcon + TextBlock)

In the first commit I also fixed the "Size to fit" function for the other columns, there the index was off by one

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 1]
```diff
@@ -625,29 +626,44 @@ private void ResizeColumnToFit(int columnToResize)
...
-				if (columnIndexFromName == -1)
-					return false;
+				return x.DesiredSize.Width;
+			}).Sum();
 
-				return columnIndexFromName == columnIndex;
-			});
+			if (stackPanels.Length >= 2)
+				mesuredSize += 4 * (stackPanels.Length - 1); // the spacing between the tags
+
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:660]: Comments should start with a capital letter

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 2]
```diff
@@ -625,29 +626,44 @@ private void ResizeColumnToFit(int columnToResize)
...
+			return MeasureTextColumnEstimate(columnIndex, measureItemsCount, maxItemLength);
+		}
+
+		private double MeasureTagColumnEstimate(int columnIndex)
 		{
-			var tbs = DependencyObjectHelpers.FindChildren<TextBlock>(FileList.ItemsPanelRoot).Where(tb =>
+			var grids = DependencyObjectHelpers.FindChildren<Grid>(FileList.ItemsPanelRoot).Where(grid => IsCorrectColumn(grid, columnIndex)).ToList();
+
+			// get the list of stack panels with the most letters
+			var stackPanels = grids
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:645]: As above

**Comment by ferrariofilippo** [comment]
Just two things:
- Can you reduce the empty space on the right?
- If you remove a tag from a file and click `Resize all columns to fit`, nothing happens. You have to change the sort order and then click it again to make it work

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 3]
```diff
@@ -625,29 +626,44 @@ private void ResizeColumnToFit(int columnToResize)
...
+			if (columnIndex == 3)
+				return MeasureTagColumnEstimate(columnIndex);
+
+			return MeasureTextColumnEstimate(columnIndex, measureItemsCount, maxItemLength);
+		}
+
+		private double MeasureTagColumnEstimate(int columnIndex)
 		{
-			var tbs = DependencyObjectHelpers.FindChildren<TextBlock>(FileList.ItemsPanelRoot).Where(tb =>
+			var grids = DependencyObjectHelpers.FindChildren<Grid>(FileList.ItemsPanelRoot).Where(grid => IsCorrectColumn(grid, columnIndex)).ToList();
```
Comment (QuaintMako) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:642]: Going to a new line as you did with the Linq request l.645 to l.649 below is the way to go to make long calls like this one clearer. Please do it here as well.

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 4]
```diff
@@ -625,29 +626,44 @@ private void ResizeColumnToFit(int columnToResize)
...
-			});
+			if (stackPanels.Length >= 2)
+				mesuredSize += 4 * (stackPanels.Length - 1); // the spacing between the tags
+
+			return mesuredSize;
+		}
+
+		private double MeasureTextColumnEstimate(int columnIndex, int measureItemsCount, int maxItemLength)
+		{
+			var tbs = DependencyObjectHelpers.FindChildren<TextBlock>(FileList.ItemsPanelRoot).Where(tb => IsCorrectColumn(tb, columnIndex));
```
Comment (QuaintMako) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:666]: Jump to new lines after each chained `.instruction`

File: src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs [hunk 5]
```diff
@@ -667,6 +683,28 @@ private double MeasureTextColumnEstimate(int columnIndex, int measureItemsCount,
...
+				"ItemType" => 8,
+				"ItemSize" => 9,
+				"ItemStatus" => 10,
+				_ => -1,
+			};
+
+			if (columnIndexFromName == -1)
+				return false;
+
+			return columnIndexFromName == columnIndex;
```
Comment (QuaintMako) [inline src/Files.App/Views/LayoutModes/DetailsLayoutBrowser.xaml.cs:705]:
```suggestion
			return columnIndexFromName != -1 && columnIndexFromName == columnIndex;
```

**Comment by yaira2** [comment]
This fixes the first part of #11295 but the double click to size is still broken.

**Comment by yaira2** [comment]
I can confirm it's working as expected. With the "Size all to fit option", does it cause the app to freeze when you use it?

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**